### ISSUE TITLE
Problem setting up AWS Batch.

### ISSUE THREAD

**Issue by MaxUlysse** [nextflow-io/nextflow#535]
Following [this post](https://www.nextflow.io/blog/2017/scaling-with-aws-batch.html) from Nextflow blog, I'm trying to set that up for CAW.

So as discussed earlier, I'm joining the files you asked for:
[nextflow.command.log](https://github.com/nextflow-io/nextflow/files/1506451/nextflow.command.log)
[task.command.log](https://github.com/nextflow-io/nextflow/files/1506452/task.command.log)
[task.command.run](https://gist.github.com/MaxUlysse/640df76a06b762343001ddf94df157a5)

the `task.command.out` and the `task.command.err` are completely empty.

I managed to correct and fix several errors before this one, but I can't go further than that, as I can't find what is wrong.

I would gladly welcome your help.

**Comment by pditommaso** [comment]
well, some little more details would help :)

**Comment by MaxUlysse** [comment]
Sorry for the issue without any description, I just pressed enter, and the issue was posted...
Writing comments at the moment

**Comment by pditommaso** [comment]
I'm wondering who is returning the following error message the log file? any idea? 

```
The user-provided path *_fastqc.{zip,html} does not exist.
```

**Comment by pditommaso** [comment]
Found, this is a bug of the `nxf_s3_upload` script. When running using the a glob pattern including curly brackets it returns this error message 

```
+ nxf_s3_upload '*_fastqc.{zip,html}' s3://cbcrg-eu/work
+ local 'pattern=*_fastqc.{zip,html}'
+ local s3path=s3://cbcrg-eu/work
+ for name in '$pattern'
+ [[ -d *_fastqc.{zip,html} ]]
+ aws s3 cp '*_fastqc.{zip,html}' 's3://cbcrg-eu/work/*_fastqc.{zip,html}' --quiet --storage-class STANDARD

The user-provided path *_fastqc.{zip,html} does not exist.
```

**Comment by MaxUlysse** [comment]
I think the problem come before that in fact.
At some point it is trying to get the files from the s3 bucket:
```
/home/ec2-user/miniconda/bin/aws s3 cp --quiet s3://caw-test-data/tiny/normal/tiny_n_L001_R1_xxx.fastq.gz tiny_n_L001_R1_xxx.fastq.gz
/home/ec2-user/miniconda/bin/aws s3 cp --quiet s3://caw-test-data/tiny/normal/tiny_n_L001_R2_xxx.fastq.gz tiny_n_L001_R2_xxx.fastq.gz
```
But no files are in the work directory in the end.
And the files are definitively in the s3 bucket.
So I'm guessing it's a problem of permissions somewhere.
But I did set up the policy `AmazonS3FullAccess` to the role `ecsInstanceRole` for my compute environment.

**Comment by pditommaso** [comment]
> But no files are in the work directory in the end.

Well, you wont see input files in the S3 work dir. They are copied in the container scratch dir.

**Comment by MaxUlysse** [comment]
Oh I see

**Comment by pditommaso** [comment]
It should be fixed. I've uploaded a new snapshot. You can git it a try b defining the following variable `NXF_VER=0.26.3-SNAPSHOT`

**Comment by MaxUlysse** [comment]
I was afraid because there was some warning in the updating NF at the beginning:
```
CAPSULE: Downloading dependency io.nextflow:nextflow:pom:0.26.3-20171128.151736-1
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.nextflow:nextflow:pom:0.26.3-20171128.151736-1 in local (file:/home/max/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency io.nextflow:nextflow:pom:0.26.3-20171128.151736-1
CAPSULE: Downloading dependency io.nextflow:nxf-commons:pom:0.26.3-20171128.151804-1
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.nextflow:nxf-commons:pom:0.26.3-20171128.151804-1 in local (file:/home/max/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency io.nextflow:nxf-commons:pom:0.26.3-20171128.151804-1
CAPSULE: Downloading dependency io.nextflow:nxf-httpfs:pom:0.26.3-20171128.151901-1
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.nextflow:nxf-httpfs:pom:0.26.3-20171128.151901-1 in local (file:/home/max/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency io.nextflow:nxf-httpfs:pom:0.26.3-20171128.151901-1
CAPSULE: Downloading dependency io.nextflow:nextflow:jar:0.26.3-20171128.151736-1
CAPSULE: Downloading dependency io.nextflow:nxf-commons:jar:0.26.3-20171128.151804-1
CAPSULE: Downloading dependency io.nextflow:nxf-httpfs:jar:0.26.3-20171128.151901-1
```
But my test is currently running on our tiny testing data set.
So thank you very much for all your help.

Where was the problem coming for in the end?

**Comment by pditommaso** [comment]
The warning are expected when downloading the a snapshot. The problem was that the glob pattern including curly brackets eg `*_fastqc.{zip,html}`

**Comment by MaxUlysse** [comment]
OK, thanks for the fix.
I also have a problem with an output file being a folder.
```
Missing output file(s) `9876T` expected by process `RunBamQC (1234-9876T)`
```
But I just fixed that by removing this process ;-).
So for the time being, I'm perfectly happy with all that.

**Comment by pditommaso** [comment]
Maybe that it could something interesting to troubleshoot !

**Comment by MaxUlysse** [comment]
Totally, but on Friday, I'm going away for a month...
So I'm afraid I won't have time to look more into it...

Just notice one more issue.

We're using this snippet to save specific files from a tools into different directories
```
publishDir '.', saveAs: { it == "${bam}.metrics" ? "${params.outDir}/${directoryMap.markDuplicatesQC}/${it}" : "${params.outDir}/${directoryMap.nonRealigned}/${it}" }, mode: 'copy'
```

But when outDir is an s3 bucket, the files are saved in the directory `s3/bucketname/...` on my own computer.

**Comment by pditommaso** [comment]
OK. Report this is a separate issue at your convenience please.

**Comment by pditommaso** [comment]
Included in version `0.26.3`.
